### ISSUE TITLE
about how to train the diffusion model?

### ISSUE THREAD

**Issue by xiaoc57** [jasonyzhang/RayDiffusion#9]
I think this is a good job. I'm a little confused about training the diffusion model. Is your training method using direct prediction x_0 to calculate the loss with the real camera or calculating epsilon and epsilon_pred to calculate the loss? I used the second one, but it was all garbage when predicting. Is there any skill?

**Comment by jasonyzhang** [comment]
We tried and got reasonable results on both. I think our performance and training stability was better when doing X_0 prediction. I would suggest trying that first. 

When doing epsilon prediction, are you using our ddpm inference code? You may need to make sure that the flag for epsilon vs X0 prediction is set properly.

**Comment by youngju-na** [comment]
@jasonyzhang Hi! sorry for commenting here if it's not appropriate. I'd like to politely ask when are you planning to release the training code :)

**Comment by ThisIsIsaac** [comment]
Hi! Is there any update to the release of the training code?

**Comment by jasonyzhang** [comment]
#10

**Comment by jasonyzhang** [comment]
The [dev branch](https://github.com/jasonyzhang/RayDiffusion/tree/dev) now has a V1 of the training code.
